### PR TITLE
GutenbergKit

### PR DESCRIPTION
**Pull request by kean**
Integrates the new block editor into the app.

There are a ton of issues, both known and unknown, but by landing this now we're able to more effectively iterate on them without getting behind `trunk`.

To test:

## Regression Notes
1. Potential unintended areas of impact
n/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a

3. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>Modules/Package.swift</code> was changed without updating its corresponding <code>Package.resolved</code>. Please resolve the Swift packages in Xcode.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift [hunk 1]
```diff
@@ -485,8 +486,13 @@ class GutenbergViewController: UIViewController, PostEditor, FeaturedImageDelega
     }
 
     func requestHTML(_ completion: @escaping () -> Void) {
+        let startTime = CFAbsoluteTimeGetCurrent()
         htmlDidChange.first()
-            .sink { completion() }
+            .sink {
+                let duration = CFAbsoluteTimeGetCurrent() - startTime
```
Comment (kean) [inline WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift:492]: This needs to be removed and the comment above.

File: WordPress/Classes/ViewRelated/Me/App Settings/ExperimentalFeaturesDataProvider.swift [hunk 2]
```diff
@@ -5,6 +5,7 @@ class ExperimentalFeaturesDataProvider: ExperimentalFeaturesViewModel.DataProvid
 
     let flags: [OverridableFlag] = [
         FeatureFlag.authenticateUsingApplicationPassword,
+//        FeatureFlag.newGutenberg,
```
Comment (kean) [inline WordPress/Classes/ViewRelated/Me/App Settings/ExperimentalFeaturesDataProvider.swift:8]: Commented out code.

File: WordPress/Classes/ViewRelated/Post/PostEditor+Publish.swift [hunk 3]
```diff
@@ -38,6 +38,11 @@ protocol PublishingEditor where Self: UIViewController {
     var debouncer: Debouncer { get }
 
     func emitPostSaveEvent()
+
+    // TODO: rework how the editors manage their changes
+    var editorHasContent: Bool { get }
```
Comment (kean) [inline WordPress/Classes/ViewRelated/Post/PostEditor+Publish.swift:43]: This needs to be reviewed – not sure if it's safe.

File: WordPress/Classes/ViewRelated/Me/App Settings/ExperimentalFeaturesDataProvider.swift (hunk 2 above)
Comment (jkmassel) [reply WordPress/Classes/ViewRelated/Me/App Settings/ExperimentalFeaturesDataProvider.swift:8]: I put this one in on purpose for when we need it 😅

**Comment by geriux** [comment]
I've tested both builds and I was able to see the current editor and also the experimental one enabling the feature flags. 🚀 

I think we could merge this as a starting point and build on or refactor it later on.

**Review by dcalhoun — COMMENTED** [review]
Testing both GBM and GBK editors worked for me. I agree it is worth merging this to avoid falling behind trunk, and optimizing it later.

**Review by geriux — APPROVED** [review]

**Pull request merged**